File: frontend/src/types/card.ts
import type { CardLabel } from "./label";

export interface CardAssigneeUser {
  user_id: string;
  name: string;
  email: string;
}

export interface CardAssignee {
  id: string;
  card_id: string;
  user_id: string;
  user: CardAssigneeUser;
}

export interface Card {
  card_id: string;
  list_id: string;
  title: string;
  description?: string;
  due_date?: string;
  position: number;
  labels?: CardLabel[];
  assignees?: CardAssignee[];
}

export interface CreateCardPayload {
  title: string;
  description?: string;
  due_date?: string;
}

export interface UpdateCardPayload {
  title?: string;
  description?: string;
  due_date?: string;
  list_id?: string;
  position?: number;
}

export interface CardsResponse {
  message: string;
  data: Card[];
}

export interface CardResponse {
  message: string;
  data: Card;
}

export interface CardAssigneeResponse {
  message: string;
  data: CardAssignee;
}

File: backend/models/card.py
import uuid
from sqlalchemy import Column, String, Text, Date, Integer, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from database import Base


class Card(Base):
    __tablename__ = "cards"

    card_id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    title = Column(String(150), nullable=False)
    description = Column(Text)
    due_date = Column(Date)
    position = Column(Integer, nullable=False)

    list_id = Column(UUID(as_uuid=True), ForeignKey("lists.list_id"), nullable=False)

    list = relationship("List", back_populates="cards")
    comments = relationship("Comment", back_populates="card")
    assignees = relationship("CardAssignee", back_populates="card")
    labels = relationship("CardLabel", back_populates="card")
